src/Translation_unit.ml

```diff
@@ -337,13 +339,15 @@ let format xunit (conf : Conf.t) ?output_file ~input_name ~source ~parsed ()
     else
       match parse xunit.parse conf ~source:fmted with
       | exception Sys_error msg -> Error (User_error msg)
-      | exception exn ->
+      | exception exn -> (
           let args =
             [("output file", dump_formatted ~suffix:".invalid-ast" fmted)]
             |> List.filter_map ~f:(fun (s, f_opt) ->
                    Option.map f_opt ~f:(fun f -> (s, String.sexp_of_t f)))
           in
-          internal_error (`Cannot_parse exn) args
+          match exn with
+          | Warning50 l -> internal_error (`Warning50 l) args
+          | exn -> internal_error (`Cannot_parse exn) args )
```

Comment:
You can define `args` (as a function or lazy) outside the match and merge the matches.

Reply:
Done.